### PR TITLE
feat: azapi 2.0.1

### LINKED ISSUE #132: [AVM Module Issue]: Unable to set selectors for overrides in policy_assignments_to_modify
### Check for previous/existing GitHub issues

- [x] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### (Optional) Module Version

v0.9.0-beta2

### (Optional) Correlation Id

_No response_

### Description

There is a mismatch between the `policy_assignments_to_modify` variable definition in this module and what's expected by the provider.
Under overrides the module asks for `selectors`:

https://github.com/Azure/terraform-azurerm-avm-ptn-alz/blob/e87b448860b92c5066b26ece52f16885e0ea9d27/variables.tf#L90-L98

However the provider wants [`override_selectors`](https://github.com/Azure/terraform-provider-alz/blob/main/docs/data-sources/architecture.md#nested-schema-for-policy_assignments_to_modifypolicy_assignmentsoverrides)

When deploying as is, selectors are ignored and it's instead set as an empty list. When the variable definition is updated to `override_selectors` it works as expected.

Although I've not tested, I suspect `resource_selectors` has the same issue as the provider documentation states it should be `resource_selector_selectors`

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
## Description

- Support for the release version of azapi 2.0
- Adds example for mgmt module
- fixes #132 

### Default

```
Apply complete! Resources: 388 added, 0 changed, 0 destroyed.
```

### Management

```
Apply complete! Resources: 421 added, 0 changed, 0 destroyed.
```

### policy assignment modification with custom lib

```
Apply complete! Resources: 250 added, 0 changed, 0 destroyed.
```

Verified all idempotent

Dependent on Azure/terraform-azurerm-avm-ptn-alz-management#74

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -277,7 +282,7 @@ map(object({
       overrides = optional(list(object({
         kind  = string
         value = string
-        selectors = optional(list(object({
+        override_selectors = optional(list(object({
```
Comment (jaredfholgate) [inline README.md:285]: Why this name change? The previous names were more succinct.

File: terraform.tf [hunk 2]
```diff
@@ -3,11 +3,11 @@ terraform {
   required_providers {
     alz = {
       source  = "azure/alz"
-      version = ">= 0.15.1, < 1.0.0"
+      version = "~> 0.16"
     }
     azapi = {
       source  = "azure/azapi"
-      version = "2.0.0-beta"
+      version = "~> 2.0, >= 2.0.1"
```
Comment (jaredfholgate) [inline terraform.tf:10]: Do we want `,< 3` here too?

File: variables.tf [hunk 3]
```diff
@@ -81,7 +81,7 @@ variable "policy_assignments_to_modify" {
       })), null)
       resource_selectors = optional(list(object({
         name = string
-        selectors = optional(list(object({
+        resource_selector_selectors = optional(list(object({
```
Comment (jaredfholgate) [inline variables.tf:84]: I preferred the old name here. It was more succinct.

We are moving from `resource_selectors.selectors` to `resource_selectors.resource_selector_selectors`, which seems like unnecessary duplication?

**Review by jaredfholgate — APPROVED** [review]
Approving, but leaving some comments to consider.

File: terraform.tf (hunk 2 above)
Comment (matt-FFFFFF) [reply terraform.tf:10]: It's most restrictive here isn't it. A logical AND?

Therefore ~> 2.0 would prevent 3.0 being selected I think.

File: README.md (hunk 1 above)
Comment (matt-FFFFFF) [reply README.md:285]: Fair. Due to an auto gen restriction on the provider but I could keep this the same.

Will revert.

File: terraform.tf (hunk 2 above)
Comment (jaredfholgate) [reply terraform.tf:10]: Yes, sorry.

File: variables.tf (hunk 3 above)
Comment (matt-FFFFFF) [reply variables.tf:84]: Changing this means re-writing the object that is fed into the provider. Think I will go with the as-is

**Pull request merged**